ext/bcmath/libbcmath/src/recmul.c

```diff
@@ -102,6 +158,47 @@ static inline void bc_fast_mul(bc_num n1, size_t n1len, bc_num n2, int n2len, bc
 	}
 }
 
+#if BC_LITTLE_ENDIAN
+# define BC_ENCODE_LUT(A, B) ((A) | (B) << 8)
+#else
+# define BC_ENCODE_LUT(A, B) ((B) | (A) << 8)
+#endif
+
+#define LUT_ITERATE(_, A) \
+	_(A, 0), _(A, 1), _(A, 2), _(A, 3), _(A, 4), _(A, 5), _(A, 6), _(A, 7), _(A, 8), _(A, 9)
+
+/* This LUT encodes the decimal representation of numbers 0-100
+ * such that we can avoid taking modulos and divisions which would be slow. */
+static const unsigned short LUT[100] = {
```

Comment:
One thing I'm wondering too is if it makes sense to make the table smaller. I.e. encode every entry in 8 bits and expand it on writing. This makes the code slightly slower but reduces cache pressure. Not sure what's best.

Reply:
In the current implementation, the size of the LUT is about 200 bytes? Even if change to 8bit, we only save 100 bytes, so I feel like we can leave it as is.

How do you think?